implementation/gate/src/main/java/eu/efti/eftigate/utils/ControlUtils.java

```diff
@@ -34,7 +34,7 @@ public static ControlDto fromGateToGateQuery(final UILQuery uilQuery, final Requ
         controlDto.setRequestId(uilQuery.getRequestId());
         controlDto.setRequestType(requestTypeEnum);
         controlDto.setStatus(StatusEnum.PENDING);
-        controlDto.setSubsetId(StringUtils.isNotBlank(uilQuery.getSubsetId()) ? uilQuery.getSubsetId() : SUBSET_ID);
+        controlDto.setSubsetId(!uilQuery.getSubsetId().isEmpty() ? uilQuery.getSubsetId().get(0) : SUBSET_ID);
```

Comment:
Just to be clear: the idea is that this line will be replaced once proper support for multiple subset ids is implemented.